code/modules/clothes_making/weaving.dm

```diff
@@ -0,0 +1,383 @@
+//TODO: support wool as well, and maybe other plants such as coton
+//I chose linen to start with because of too much time playing Pharaoh
+/obj/structure/spinning_wheel
+	name = "spinning wheel"
+	desc = "Allows you to manually weave grown flax into linen cloth."
+	icon = 'icons/obj/clothes_making.dmi'
+	icon_state = "wooden_loom"
+	density = 1
+	anchored = 0
+	pass_flags_self = PASSMACHINE
+
+	var/remaining_cloth_to_spin = 0
+	var/mob/spinner = null
+
+/obj/structure/spinning_wheel/Destroy()
+	if (spinner)
+		spinner = null
+		processing_objects.Remove(src)
+	..()
+
+/obj/structure/spinning_wheel/examine(mob/user)
+	..()
+	if(remaining_cloth_to_spin > 0)
+		to_chat(user, "<span class='info'>There is enough flax in it to produce [remaining_cloth_to_spin] more cloth sheets.</span>")
+	else
+		to_chat(user, "<span class='info'>Grow some flax then put it in there before you can use it.</span>")
+
+/obj/structure/spinning_wheel/attack_hand(var/mob/user)
+	if (spinner)
+		if (user == spinner)
+			to_chat(user, "<span class='notice'>You interrupt your weaving.</span>")
+			spinner = null
+			processing_objects.Remove(src)
+			update_icon()
+		else
+			to_chat(user, "<span class='warning'>\The [spinner] is currently weaving at this [src] already.</span>")
+	else if (remaining_cloth_to_spin > 0)
+		spinner = user
+		to_chat(user, "<span class='notice'>You start weaving the flax into cloth.</span>")
+		processing_objects.Add(src)
+		playsound(src, 'sound/machines/loom_wooden_start.ogg', 10, 0)
+		update_icon()
+	else
+		to_chat(user, "<span class='warning'>Can't use the wheel before some flax to weave has been added to it.</span>")
+
+/obj/structure/spinning_wheel/attackby(var/obj/item/W, var/mob/user)
+	if(W.is_wrench(user))
+		W.playtoolsound(src, 100)
+		user.visible_message("<span class='notice'>[user] starts disassembling \the [src].</span>", \
+		"<span class='notice'>You start disassembling \the [src].</span>")
+		if(do_after(user, src, 20))
+			user.visible_message("<span class='warning'>[user] dissasembles \the [src].</span>", \
+			"<span class='notice'>You dissasemble \the [src].</span>")
+			var/turf/T = get_turf(src)
+			new /obj/item/stack/sheet/wood(T, 10)
+			for (var/i = 1 to round(remaining_cloth_to_spin/CLOTH_PER_FLAX))
+				new /obj/item/weapon/reagent_containers/food/snacks/grown/flax(T)
+			qdel(src)
+	else if (istype(W, /obj/item/weapon/reagent_containers/food/snacks/grown/flax))
+		if(user.drop_item(W, loc))
+			remaining_cloth_to_spin += CLOTH_PER_FLAX
+			qdel(W)
+			to_chat(user, "<span class='notice'>You prepare the flax to be weaved by the wheel.</span>")
+			playsound(src, 'sound/items/bonegel.ogg', 50, 0)
+			update_icon()
+	else if (istype(W, /obj/item/weapon/storage/bag/plants))
+		var/inserted = FALSE
+		for (var/obj/item/weapon/reagent_containers/food/snacks/grown/flax/F in W.contents)
+			remaining_cloth_to_spin += CLOTH_PER_FLAX
```

Comment:
seems like it might be nice to scale on robustness or some other attribute but this is fine

Reply:
it will scale on upgrades when using the electric loom. I intend the spinning loom to produce only as much as an un-upgraded machine will. (and upgraded machines can already produce a LOT of cloth).

Botanists may prefer to upgrade the plant's yield over its potency.

Higher potency however means more flax oil when ground.